Sources/SpeziOnboarding/OnboardingFlow/OnboardingNavigationPath.swift

```diff
@@ -162,13 +165,34 @@ public class OnboardingNavigationPath {
     /// - Parameters:
     ///   - customView: A custom onboarding `View` instance that should be shown next in the onboarding flow.
     ///     It isn't required to declare this view within the ``OnboardingStack``.
-    public func append(customView: any View) {
-        let customOnboardingStepIdentifier = OnboardingStepIdentifier(fromView: customView, custom: true)
+    public func append<V: View>(customView: V) {
+        let customOnboardingStepIdentifier = OnboardingStepIdentifier(
+            onboardingStepType: String(describing: V.self),
+            custom: true
+        )
         customOnboardingSteps[customOnboardingStepIdentifier] = customView
         
         appendToInternalNavigationPath(of: customOnboardingStepIdentifier)
     }
-    
+
+    /// Moves the navigation path to the custom `Identifiable` view.
+    ///
+    /// - Note: The custom `View` does not have to be declared within the ``OnboardingStack``.
+    ///     Resulting from that, the internal state of the ``OnboardingNavigationPath`` is still referencing to the last regular `OnboardingStep`.
+    ///
+    /// - Parameters:
+    ///   - customView: A custom onboarding `View` instance that should be shown next in the onboarding flow.
+    ///     It isn't required to declare this view within the ``OnboardingStack``.
+    public func append<V: View & Identifiable>(customView: V) {
+        let customOnboardingStepIdentifier = OnboardingStepIdentifier(
+            onboardingStepType: String(describing: customView.id),
```

Comment:
Why are we turning the `Hashable` id into a String here? There is no guarantee that the String representation of two Hashables that have the same hash is the same.
I would propose to change the implementation of `OnboardingStepIdentifier`. Making `onboardingStepType` to store the hash of the id of a given onboarding view (maybe renaming the property would also make sense to reflect the new semantics).

Refer to the documentation of [`Hasher`](https://developer.apple.com/documentation/swift/hasher) on how to calculate the hash of a `Hashable`.